cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryAdapterConfigurator.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.deploy;
+
+import co.cask.cdap.api.templates.ApplicationTemplate;
+import co.cask.cdap.app.ApplicationSpecification;
+import co.cask.cdap.app.program.Archive;
+import co.cask.cdap.app.program.ManifestFields;
+import co.cask.cdap.common.lang.jar.BundleJarUtil;
+import co.cask.cdap.common.utils.DirUtils;
+import co.cask.cdap.proto.AdapterConfig;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.templates.AdapterSpecification;
+import co.cask.cdap.templates.DefaultAdapterConfigurer;
+import com.google.common.base.Preconditions;
+import com.google.common.io.Files;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gson.Gson;
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.jar.Manifest;
+
+/**
+ * In Memory Configurator doesn't spawn a external process, but
+ * does this in memory.
+ *
+ * @see SandboxConfigurator
+ */
+public final class InMemoryAdapterConfigurator {
+  private static final Gson GSON = new Gson();
+  private static final Logger LOG = LoggerFactory.getLogger(InMemoryAdapterConfigurator.class);
+  /**
+   * JAR file path.
+   */
+  private final Location archive;
+  private final AdapterConfig adapterConfig;
+  private final ApplicationSpecification templateSpec;
+  private final String adapterName;
+
+  public InMemoryAdapterConfigurator(Id.Namespace id, Location archive, String adapterName,
+                                     AdapterConfig adapterConfig, ApplicationSpecification templateSpec) {
+    Preconditions.checkNotNull(id);
+    Preconditions.checkNotNull(archive);
+    this.archive = archive;
+    this.adapterConfig = adapterConfig;
+    this.templateSpec = templateSpec;
+    this.adapterName = adapterName;
+  }
+
+  /**
+   * Executes the <code>Application.configure</code> within the same JVM.
+   * <p>
+   * This method could be dangerous and should be used only in standalone mode.
+   * </p>
+   *
+   * @return A instance of {@link ListenableFuture}.
```

Comment:
Doesn't return a ListenableFuture. Related to earlier comment. if we are not expecting to implement configurator fix docs.